### ISSUE TITLE
feat(accessibility): Update the BulkRelationDialog to be a more accessible design

### ISSUE THREAD

**Issue by scarletmerlin123** [OpenCTI-Platform/opencti#17889]
## Use case

The current dialog presents multiple issues related to accessibility:

- The screen reader doesn't state the entity that the relationship is from, making it harder for users to have appropriate context.
- The inputs are not labeled so users do not know what they are for
- The added items are not contained in a grid so keyboard users must tab through all inputs to navigate the list of added relationships
- No helper text or validation text is provided to alert users that they must create missing entities before submitting
- No helper text to clarify that users need to separate entities on new lines in the text area input 
- After creating missing entities focus resets and forces the user to navigate through all inputs again before pressing create
- The horizontal orientation of the form causes issues when using browser zoom or when using smaller screens

The layout should be revised to resolve these issues.
## Current workaround

None

## Proposed solution

Rearrange the layout to be vertically oriented and resolve issues above.

- Remove the use of the card display for the from entity and replace with accessible text
- Label all inputs
- add helper text to the "To" text area to clarify expected input
- Add visually persistent create missing entities button below the text area that becomes enabled if there are entities not in the platform.
- Resolve focus reset after opening create missing entities dialog
- Add form validation error if missing entities haven't been created
- add aria-live="polite" attribute to the created relationship section because it dynamically populates as users type
- Update layout of created relationship section to use a table/grid structure to allow labeling of inputs based on the column and allow easier keyboard navigation

## Additional information

Rough mockup:

<img width="762" height="648" alt="Image" src="https://github.com/user-attachments/assets/db81de1c-301f-4428-8d58-c079902d9e97" />

## If the feature request is approved, would you be willing to submit a PR?

Yes
